src/features/dashboard/USDCMarketSlice.ts

```diff
@@ -0,0 +1,484 @@
+    const degenUSDC = new Contract(testnet_addresses.degenUSDC, ERC20Immutable.abi, ethersProvider);
+
+    try {
+        const  borrowRateMantissa = await degenUSDC.borrowRatePerBlock();
+        const decimals = await degenUSDC.decimals()
+        const borrowRate = formatUnits(borrowRateMantissa, decimals);
+        console.log(`[Console] successfully called on thunk 'updateBorrowRate'`);
+
+        return Number(borrowRate);
+    } catch (error) {
+        console.log(`[Console] an error occured on thunk 'updateBorrowRate': ${error}`)
+        return 0;
+    }
+});
+
+// Activities
+
+///////////  Supply Market Thunks
+export const approveUSDC = createAsyncThunk('usdc/Approve', async ({ amount, addressToApprove }: { amount: number, addressToApprove: string }) => {
+    
+    const [wallet] = onboard.state.get().wallets;
+
+    if (wallet === undefined) {
+        return 0;
+    }
+
+    console.log(`[Console] initiating thunk, 'approveUSDC' ...`);
+
+    
+    let ethersProvider = new ethers.BrowserProvider(wallet.provider, 'any');
+    const signer = await ethersProvider.getSigner();
+    const USDC = new Contract(testnet_addresses.USDC, ERC20.abi, signer);
+    const spender = testnet_addresses.degenUSDC;
+    try {
+        let tx = await USDC.approve(spender, amount);
+        await tx.wait();
+        console.log(`[Console] successfully called on thunk 'approveUSDC'`);
+    } catch (error) {
+        console.log(`[Console] an error occurred on thunk 'approveUSDC': ${error} `)
+    }
+
+})
+
+///////////  Supply Market Thunks
+
+export const supplyUSDC = createAsyncThunk('usdc/Supply', async () => {
+    console.log(`[Console] initiating thunk, 'supplyUSDC' ...`);
+
+    const [wallet] = onboard.state.get().wallets;
+
+    if (wallet === undefined) {
+        return 0;
+    }
+
+    console.log(`[Console] initiating thunk, 'supplyUSDC' ...`);
+
+    
+    let ethersProvider = new ethers.BrowserProvider(wallet.provider, 'any');
+    const signer = await ethersProvider.getSigner();
+
+    const degenUSDC = new Contract(testnet_addresses.degenUSDC, ERC20Immutable.abi, signer);
+    const supplyAmount = 1 * 1e18;
+
+    try {
+        const tx = await degenUSDC.mint(supplyAmount);
+        tx.wait();
+        console.log(`[Console] successfully called on thunk 'supplyUSDC'`);
+    } catch (error) {
+        console.log(`[Console] an error occurred on thunk 'supplyUSDC': ${error} `)
+
+    }
+})
+
+export const withdrawUSDC = createAsyncThunk('usdc/withdraw', async () => {
+    
+    const [wallet] = onboard.state.get().wallets;
+
+    if (wallet === undefined) {
+        return 0;
+    }
+    
+    console.log(`[Console] initiating thunk, 'withdrawUSDC' ...`);
+
+
+    let ethersProvider = new ethers.BrowserProvider(wallet.provider, 'any');
+    const signer = await ethersProvider.getSigner();
+    const degenUSDC = new Contract(testnet_addresses.degenUSDC, ERC20.abi, signer);
+
+    try {
+
+        degenUSDC.redeemUnderlying(1)
```

Comment:
[Question]

Why is the value for here 1?

Reply:
placeholder